From 16f035553b56ea6bb285745194eb6a52d46a8861 Mon Sep 17 00:00:00 2001
From: Oliver Heger <oheger@apache.org>
Date: Sun, 14 Mar 2010 14:30:36 +0000
Subject: [PATCH] CONFIGURATION-408: Switched to commons-lang 2.5. Added a unit
 test to ensure that with commons-lang 2.5 the escaping problem does not
 occur.

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/configuration/branches/configuration2_experimental@922865 13f79535-47bb-0310-9956-ffa450edef68
---
 pom.xml                                            |  2 +-
 .../TestPropertiesConfiguration.java               | 14 ++++++++++++++
 xdocs/changes.xml                                  |  5 +++++
 xdocs/dependencies.xml                             |  2 +-
 4 files changed, 21 insertions(+), 2 deletions(-)

diff --git a/pom.xml b/pom.xml
index 38aa74635e..6660b78e6b 100644
--- a/pom.xml
+++ b/pom.xml
@@ -205,7 +205,7 @@
     <dependency>
       <groupId>commons-lang</groupId>
       <artifactId>commons-lang</artifactId>
-      <version>2.4</version>
+      <version>2.5</version>
     </dependency>
 
     <dependency>
diff --git a/src/test/java/org/apache/commons/configuration2/TestPropertiesConfiguration.java b/src/test/java/org/apache/commons/configuration2/TestPropertiesConfiguration.java
index a32f262162..f417f5d03c 100644
--- a/src/test/java/org/apache/commons/configuration2/TestPropertiesConfiguration.java
+++ b/src/test/java/org/apache/commons/configuration2/TestPropertiesConfiguration.java
@@ -976,6 +976,20 @@ public void testKeepSeparators() throws ConfigurationException, IOException
                 separatorTests.length, foundLines.size());
     }
 
+    /**
+     * Tests whether properties with slashes in their values can be saved. This
+     * test is related to CONFIGURATION-408.
+     */
+    public void testSlashEscaping() throws ConfigurationException
+    {
+        conf.setProperty(PROP_NAME, "http://www.apache.org");
+        StringWriter writer = new StringWriter();
+        conf.save(writer);
+        String s = writer.toString();
+        assertTrue("Value not found: " + s, s.indexOf(PROP_NAME
+                + " = http://www.apache.org") >= 0);
+    }
+
     /**
      * Creates a configuration that can be used for testing copy operations.
      *
diff --git a/xdocs/changes.xml b/xdocs/changes.xml
index 733d7ca848..0019840132 100644
--- a/xdocs/changes.xml
+++ b/xdocs/changes.xml
@@ -83,6 +83,11 @@
         HierarchicalINIConfiguration now correctly saves sections whose name
         contains delimiter characters.
       </action>
+      <action dev="oheger" type="update" issue="CONFIGURATION-408">
+        PropertiesConfiguration.save() escaped slashes in properties values.
+        This was caused by a bug in commons-lang 2.4. Updating to the new
+        version commons-lang 2.5 fixed this problem.
+      </action>
       <action dev="oheger" type="fix" issue="CONFIGURATION-407">
         Fixed a potential IllegalStateException in HierarchicalINIConfiguration
         that can be thrown when the global section is requested concurrently.
diff --git a/xdocs/dependencies.xml b/xdocs/dependencies.xml
index c4e51a053e..13ac60d5ec 100644
--- a/xdocs/dependencies.xml
+++ b/xdocs/dependencies.xml
@@ -86,7 +86,7 @@
                   <tbody>
                     <tr>
                       <td>commons-lang</td>
-                      <td>2.2, 2.3</td>
+                      <td>2.2, 2.3, 2.4, 2.5</td>
                     </tr>
                     <tr>
                       <td>commons-beanutils</td>
